cdap-unit-test/src/test/java/co/cask/cdap/partitioned/PartitionRollbackTestRun.java

```diff
@@ -377,14 +377,11 @@ public void run() {
 
     // 14. multi-output mapreduce writing partition fails because partition already exists
     // 14a. first output fails, other output must rollback partition 5
-    // TODO: bring this back when CDAP-8766 is fixed
```

Comment:
this JIRA is actually unresolved...

Reply:
Yes, you fixed all these cases with the refactoring in 4.3.x, but you must have used a different Jira. So all that remains for this one is to re-enable the test cases. They all pass now.

Reply:
I read the JIRA, but am not sure that those cases are indeed fixed.
Is it because the MapReduce job doesn't actually run if the output files exist, and so the PFS's `onFailure` doesn't execute in this case?

Reply:
In some cases, yes. There is a checkOutputSpec() or similar in the output format that, if that fails, then the MR does not start.

With dynamic partitioning, it will execute because it does not know ahead of time what partitions can exist.